src/shared/components/enhancedReactTable/EnhancedReactTable.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import {Table as DataTable} from "reactableMSK";
+import * as _ from 'underscore';
+import {IEnhancedReactTableProps, IColumnDefMap} from "IEnhancedReactTableProps";
+import {IColumnFormatterData} from "./IColumnFormatterProps";
+
+/**
+ * @author Selcuk Onur Sumer
+ */
+export default class EnhancedReactTable extends React.Component<IEnhancedReactTableProps, {}>
+{
+    constructor(props:IEnhancedReactTableProps)
+    {
+        super(props);
+        this.state = {};
+    }
+
+    public render() {
+        let {
+            reactTableProps,
+            columns,
+            rawData
+        } = this.props;
+
+        columns = columns || {};
+
+        // update (override) react table props
+        reactTableProps.data = this.convertToTableData(columns, rawData);
+        reactTableProps.columnFormatters = this.columnFormatters(columns);
+        reactTableProps.sortable = this.resolveSortable(columns);
+        reactTableProps.filterable = this.resolveFilterable(columns);
+
+        // TODO future work: column order, column visibility
+
+        return(
+            <DataTable {...reactTableProps} />
+        );
+    }
+
+    private resolveSortable(columns:IColumnDefMap):Array<any>
+    {
+        let sortable:Array<any> = []; // Array<string|function>
+
+        _.each(columns, function(column:any) {
+            if (_.isFunction(column.sortable)) {
+                sortable.push({
+                    column: column.name,
+                    sortFunction: column.sortable
+                });
+            }
+            else if (column.sortable) {
+                sortable.push(column.name);
+            }
+        });
+
+        return sortable;
+    }
+
+    private resolveFilterable(columns:IColumnDefMap):Array<any>
+    {
+        let filterable:Array<any> = []; // Array<string|function>
+
+        _.each(columns, function(column:any) {
+            if (_.isFunction(column.filterable)) {
+                filterable.push({
+                    column: column.name,
+                    filterFunction: column.filterable
+                });
+            }
+            else if (column.filterable) {
+                filterable.push(column.name);
+            }
+        });
+
+        return filterable;
+    }
+
+    private columnFormatters(columns:IColumnDefMap):any
+    {
+        let formatters:any = {};
+
+        _.each(columns, function(column:any) {
+            formatters[column.name] = column.formatter;
+        });
+
+        return formatters;
+    }
+
+    private convertToTableData(columns:IColumnDefMap, rawData:Array<any>):Array<any>
+    {
+        let rows:Array<any> = [];
+
+        _.each(rawData, function(element:any) {
+            let row:any = {};
+
+            _.each(columns, function(columnDef:any) {
```

Comment:
I think maybe a small change to MSKReactable would make this much easier.   The column formatter component should be passed these three properties automatically: tableData, rowData, and columnData.   That way you don't have to do all this work to put references to the parent data structures in each cell data object.  I'm happy to make this change if you agree that it makes sense.